Expensify/App
# Updated Amount validation conditions to not allow indefinite zeroes at the start.

### Details
The IOU Amount page didn't impose limits on the length of the string if zeroes are appended to the string. Hence the string can be indefinitely long if zeroes are used at the start.

### Fixed Issues

$ https://github.com/Expensify/App/issues/7068

### Tests

1. Open IOU Amount Page(Send/Receive).
2. Enter amounts starting with zeroes.
3. Test edge cases such as 0's at end and fractional values.



### QA Steps

1. Try entering a string of consecutive zeroes.
2. Test if fractional values appear as expected.


### Tested On


### Screenshots


#### Web



https://user-images.githubusercontent.com/48352577/148817443-6f405a07-c6d9-4818-ac4d-a769734e3683.mov



#### Mobile Web

https://user-images.githubusercontent.com/48352577/148817462-0dc8efb4-4d80-45b5-b664-054e3647ac7c.mp4


#### Desktop


#### iOS


#### Android


https://user-images.githubusercontent.com/48352577/148826431-5ec39a0a-9d9f-49b8-b5cd-4511f94c94e8.mp4
